domain/credential/state/types.go

```diff
@@ -180,6 +186,18 @@ type credentialKey struct {
 	OwnerName      string `db:"owner_name"`
 }
 
+// modelCredentialStatus represents the cloud credential key and validity status
+// of the credential that is being used by a model. The members of this type
+// are considered optional with sql null types. If the values are null it
+// indicates that the model has no cloud credential set and that the cloud
+// supports empty auth type.
+type modelCredentialStatus struct {
```

Comment:
Would it be safer that instead of this, we check the auth_type of the cloud through the usage of cloud_name to determine if the cloud support empty auth type?

Reply:
We will only allow setting of a null credential if the cloud supports it. In this case we aren't interested if the model's cloud supports the value but that if a credential of type empty auth is being used. A null set credential on the model indicates this.

Reply:
I see, thanks for the clarification tom! Perhaps the concern was that because we indicate it to be invalid as long as one of these fields is null, and im wondering if there would ever be a case where credentialName and CloudName is not null, but owner name is, and if we still constitute this as invalid.

Reply:
Yeah good question. There should never be a case where one field would be null and the rest not. We explicitly don't allow this in the DDL. A credential's natural key is comprised of those 3 values and we don't allow combinations of them being set.